docs/guides/relay.md

```diff
@@ -0,0 +1,381 @@
+meaning that the slice on it will translate to a `LIMIT`/`OFFSET` in the SQL
+query, which will fetch only the data that is needed from the database.
+
+Also note that if that object doesn't have a `__getitem__` attribute, it will
+use `itertools.islice` to paginate it, meaning that when a generator is being
+resolved it will only generate as much results as needed for the given pagination,
+the worst case scenario being the last results needing to be returned.
+
+</Note>
+
+You may want to use a different approach to paginate your results. For example,
+a cursor-based approach. For that you need to subclass the `Connection` type
+and implement your own `from_nodes` method. For example, suppose that in our
+exaple above, we want to use the fruit's weight as the cursor, we can implement
+it like that:
+
+```python
+from strawberry.relay import to_base64
+
+
+@strawberry.type
+class FruitCustomPaginationConnection(strawberry.relay.Connection[Fruit]):
+    @classmethod
+    def from_nodes(
+        cls,
+        nodes: Iterable[Fruit],
+        *,
+        info: Optional[Info] = None,
+        total_count: Optional[int] = None,
+        before: Optional[str] = None,
+        after: Optional[str] = None,
+        first: Optional[int] = None,
+        last: Optional[int] = None,
+    ):
+        # Note that this is a showcase implementation and is far from
+        # being optimal performance wise
+        edges_mapping = {
+            to_base64("fruit_name", n.name): strawberry.relay.Edge(
+                node=n,
+                cursor=to_base64("fruit_name", n.name),
+            )
+            for n in sorted(nodes, key=lambda f: f.name)
+        }
+        edges = list(edges_mapping.values())
+        first_edge = edges[0] if edges else None
+        last_edge = edges[-1] if edges else None
+
+        if after is not None:
+            after_edge_idx = edges.index(edges_mapping[after])
+            edges = [e for e in edges if edges.index(e) > after_edge_idx]
+
+        if before is not None:
+            before_edge_idx = edges.index(edges_mapping[before])
+            edges = [e for e in edges if edges.index(e) < before_edge_idx]
+
+        if first is not None:
+            edges = edges[:first]
+
+        if last is not None:
+            edges = edges[-last:]
+
+        return cls(
+            edges=edges,
+            page_info=strawberry.relay.PageInfo(
+                start_cursor=edges[0].cursor if edges else None,
+                end_cursor=edges[-1].cursor if edges else None,
+                has_previous_page=(
+                    first_edge is not None and bool(edges) and edges[0] != first_edge
+                ),
+                has_next_page=(
+                    last_edge is not None and bool(edges) and edges[-1] != last_edge
+                ),
+            ),
+        )
+
+
+@strawberry.type
+class Query:
+    fruits: FruitCustomPaginationConnection
```

Comment:
this is neat, but it will prevent to instantiate the type (query in this case, but can be another object type) without passing `fruits` :)

a workaround I found is this, but it's not nice:

```python
@strawberry.type
class MyObjectType:
    fruits: FruitCustomPaginationConnection = None
```

Reply:
I don't know if there's a solution for this other than maybe allowing:

```python
@strawberry.type
class MyObjectType:
    fruits: FruitCustomPaginationConnection = strawberry.relay.connection()
```